src/modules/search/components/OmniSearch.tsx

```diff
@@ -0,0 +1,136 @@
+
+  const { data, loading } = useOmniSearchClientsQuery({
+    variables: { input: { textSearch: value } },
+    skip: !value,
+  });
+
+  const clients = data?.clientSearch?.nodes || [];
+
+  const values = useAutocomplete({
+    id: 'omnisearch',
+    options: data?.clientSearch?.nodes || [],
+    filterOptions: (x) => x,
+    groupBy: (option) => option.__typename || 'other',
+    getOptionLabel: (option) => option.firstName || option.id,
+    onInputChange: (_e, value) => setValue(value),
+    inputValue: value || '',
+    clearOnBlur: false,
+  });
+
+  return (
+    <Box {...values.getRootProps()}>
+      <TextInput
+        inputProps={values.getInputProps()}
+        placeholder='Search'
+        size='small'
+        InputProps={{
+          startAdornment: <SearchIcon color='disabled' />,
+          ref: values.setAnchorEl,
+        }}
+      />
+      <Popper
+        open={values.popupOpen}
+        anchorEl={values.anchorEl}
+        placement='bottom-start'
+      >
+        <Paper
+          sx={{
+            p: 2,
+            minWidth: values.anchorEl?.getBoundingClientRect()?.width,
+          }}
+        >
+          {loading ? (
+            <Box
+              sx={{
+                display: 'flex',
+                justifyContent: 'center',
+                alignItems: 'center',
+                py: 2,
+                color: 'text.disabled',
+              }}
+            >
+              <CircularProgress color='inherit' />
+            </Box>
+          ) : (
+            <Grid container spacing={2}>
+              {!value ? (
+                <Grid item xs={12}>
+                  <Typography color='text.disabled'>
+                    Search for clients and/or projects
+                  </Typography>
+                </Grid>
+              ) : (
+                <>
+                  {[
+                    ['Client', 'Clients'],
+                    ['Project', 'Projects'],
+                  ].map(([key, label]) => {
+                    const options = flatten(
+                      (
+                        values.groupedOptions as AutocompleteGroupedOption<
+                          NonNullable<typeof clients>[number]
+                        >[]
+                      )
+                        .filter((opt) => opt.group === key)
+                        .map((g) => g.options)
+                    );
+
+                    return (
+                      <Grid item key={key}>
+                        <Typography variant='overline'>{label}</Typography>
+                        {isEmpty(options) ? (
+                          <Typography color='text.disabled' variant='body2'>
+                            No {label} found
+                          </Typography>
+                        ) : (
+                          options.map((option) => {
+                            return (
+                              <div key={option.id}>
+                                <MenuItem
```

Comment:
I'm not able to tab through the menu items. Can we update so it is keyboard navigable?

Reply:
Sure, I'll take a look and see if we can do that. There might be some stuff in the `useAutocomplete` hook I can use

Reply:
@gigxz This is fixed. it doesn't actually do tab navigation, as that would make the autocomplete behave poorly, but the arrow key navigation now works as expected. I also cleaned up some of the routing behavior and added an indication if we're currently looking at a given client in the list. Let me know if you need me to do more here

Reply:
Nice. This looks good, I'll merge and deploy! Then we will hopefully get some of Ash's time this week to give feedback on the interactions and UI.